docs/dcp2_system_design.rst

```diff
@@ -1199,6 +1222,72 @@ for a project with one non-stratified matrix.
    that file may have a more up-to-date list
 
 
+Describing CGMs as analysis files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The mechanism defined in `Describing CGMs as supplementary files`_ is
+deprecated. Instead of representing a contributor-generated matrix (CGM) as a
+stand-alone supplementary file, it is described more accurately as an
+``analysis_file`` entity that is an output of an ``analysis_process`` entity.
+
+Information about the provenance of the CGMs for a particular project may be
+limited. The DCP/2 may not be in possession of intermediate artifacts like the
+sequence data or any information about those artifacts because, for example,
+they are subject to access restrictions the DCP/2 does not yet have the
+capabilities to enforce. Even if the DCP/2 is in possession of intermediate
+artifacts and knows their provenance, their bearing on the contributed
+matrices may not have been supplied.
+
+Either way, the subgraph(s) containing CGM analysis files may need to be an
+approximation. If they are, the ``analysis_process`` entity that lists one or
+more CGMs as its output will not have any sequence files or other intermediate
+matrice files as inputs — as they would in the case of `DCP/2-generated
+matrices`_ — but instead directly refer to biomaterial entities like
+``specimen_from_organism``, ``cellline`` or ``organoid`` as its input,
+skipping the sequence files and any intermediate matrices.
+
+Consequently, the Azul indexer extends its definition of *sample* to include
+not only biomaterials that are used as sequencing inputs but also any
+biomaterials linked directly to such an approximate `analysis_process` entity.
```

Comment:
tl;dr: Based on my understanding of how Azul currently creates 'samples', I think this definition could result in varying definitions of a 'sample' depending on how we model an experiment, even if the real world processes of the experiment are the same, which I am not sure is ideal. 

I believe Azul currently considers the 'sample' the input of the input to the `sequencing_file`, but the above sentence says samples are 'sequencing inputs'. Are we proposing to change the definition of a 'sample' to the direct input before either a `sequencing_file` or an `analysis_file` (if directly linked to a biomaterial)? If I have understood this correctly I go into more detail below, if I haven't, then disregard.

For example if we consider a standard 10X primary tissue experiment with
`donor_organism`->`specimen_from_organism`->`cell_suspension`->`sequencing_file`->`analysis_file(CGM)`
In this case, I thought Azul considered the `specimen_from_organism` to be the sample.
Whereas if we consider a managed access experiment, that has the same real world experimental structure (10X primary tissue), but we don't have access to the sequencing files so don't explicitly model them, it would be 
`donor_organism`->`specimen_from_organism`->`cell_suspension`->`analysis_file(CGM)`
and based on a sample being 'directly linked to the `analysis_process`', the `cell_suspension` would be classified as the 'sample'. 

I am not sure if it is possible to have a simple linking rule to define a 'sample', avoid an inconsistent definition, while maintaining the flexibility to link any biomaterial to an `analysis_file`, so maybe we just accept this will happen, but wanted to point it out anyways since it might confuse consumers. In the browser the sample type field is described as 'the type of biomaterial used to create the cell suspension' so maybe we need to redefine that, or use some kind of more complex logic like, if there is a cell suspension, the sample is the input to the cell suspension, if there isn't a cell suspension, the sample is the biomaterial input to the first file in the subgraph (whether sequence or analysis). 

I also don't think it is correct to insinuate that a biomaterial that is linked directly to an `analysis_process` is not a sequencing input, since the `analysis_process` would often be described by a `sequencing_protocol` and in the real world probably was a sequencing input. The difference here is that it doesn't result in a `sequencing_file` as an output. Perhaps something like 'not only biomaterials that result in a `sequencing_file` as an output, but also any biomaterials...'

Reply:
What I wrote is not correct and I will fix it. A sample as currently defined in Azul is the nearest biomaterial ancestor of a sequence_file that is not a cell suspension. This is the definition that was agreed on with EBI folks in the room at a F2F in Santa Cruz. 

Sequencing input is currently defined as any biomaterial that is an input to a process with a sequencing protocol. Usually that's a cell suspension but I was told that it sometimes can be a liquid specimen like a drop of blood. I think we may need to adjust that definition, too. I think we simply don't know what the sequencing input is in these approximate graphs.

So let me make those changes and we take it from there, ok?

Reply:
Addressed.